docs/block-processing/us.md

```diff
@@ -148,55 +218,50 @@ data BlockVersionModifier = BlockVersionModifier
+### Block version life cycle
 
-Formally, let's say a proposal became _confirmed_ in `s` epoch and current epoch is `t`:  
-if portion of block issuers' stakes, which issued blocks of this `BlockVersion` at least once, is greater than
-`max spMinThd (spInitThd - (t - s) * spThdDecrement)`, then proposal's `BlockVersion` becomes **adopted**.
+![Block version states](us-bv-states.png)
 
-Intuitively, the threshold which needed to adopt `BlockVersion` is decreasing in each epoch but cannot become
-less than some reasonable minimal value (`spMinThd`).
+Once an update proposal which introduces a protocol update becomes _confirmed_, corresponding block version may have one of following states:
 
-So we check this rule at the beginning of each epoch for each _confirmed_ proposal 
-and adopt one of block version from _confirmed_ proposals if this version satisfies the rule.
+* Competing
+  * Block version which is candidate to be *adopted*
+* Adopted
+  * Current version of protocol
+* Never to become Adopted
 
+At any point in time, only one version is considered *adopted* by blockchain. This version is used to validate all blocks coming to blockchain. List of block versions which were at some point considered *adopted* form a linear order on `(Maj, Min)` pairs.
 
 #### Competing block version
+
 Block version is called **competing** if it may become adopted and 
 there is a confirmed proposal with this block version.
-There are rules defining whether block version may become adopted after current last adopted version.  
-The following rules regarding major and minor versions take place:
-  * The proposed major version must be equal to or greater by `1` last adopted one.
-  * If the proposed major version is greater than last adopted one by `1`, then minor version must be `0`.
-  * If the proposed major version is equal to the last adopted one, then minor version 
-    can be either same as the last adopted one or greater by `1`.
 
-Also the following rules regarding alternative versions take place (assuming checks above passed):
-  * If `(Major, Minor)` of proposed version is equal to `(Major, Minor)` of
-    last adopted version, then alternative version must be equal to alternative version of last adopted version.
-  * Otherwise `(Major, Minor)` of proposed version is lexicographically greater 
-    than `(Major, Minor)` of last adopted version. In this case
-    other proposed block versions with same `(Major, Minor)` are considered. 
-    Let's call set of these block version `X`:
-    * If `X` is empty, given alternative version must be `0`.
-    * Otherwise it must be in `X` or greater than maximum from `X` by `1`.
+When some block version becomes *adopted*, all other *competing* block versions are never to become *adopted*.
+
+Note, that at any point in time all *competing* block versions have same major and minor versions (i.e. differ only in alt version).
```

Comment:
this is also wrong. we can have such set of competing versions `{1.0.1, 1.1.0}`, when `adopted = 1.0.0`

Reply:
so it clarifies why your previous sentence 
>When some block version becomes *adopted*, all other *competing* block versions are never to become *adopted*.

is wrong: we can adopt the former and then the latter,